app/models/user.rb

```diff
@@ -7,40 +7,32 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   devise :omniauthable, omniauth_providers: %i[facebook]
 
-  has_many :requested_friendships, class_name: 'Friendship',
-            foreign_key: 'requestor_id', dependent: :destroy
-  has_many :received_friendships, class_name: 'Friendship',
-            foreign_key: 'acceptor_id', dependent: :destroy
-
-  has_many :posts, -> {includes(:likes, :comments).order(id: :desc)}, dependent: :destroy
+  has_many :posts, dependent: :destroy
+  has_many :comments, dependent: :destroy
   has_many :notifications, dependent: :destroy, foreign_key: 'receiver_id'
 
```

Comment:
what is the relationship between the user model and the like model

Reply:
Hi!
Thank you for the review. We decided to link the like model to the post or comment it belongs to.
It uses polymorphic via subject association. We still can add has_many likes to the user but nowhere we use that in the code. Something like: `subject.likes.build()` is used instead.